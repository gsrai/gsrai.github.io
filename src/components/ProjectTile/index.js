import React from 'react'
import PropTypes from 'prop-types'

export default class ProjectTile extends React.Component {
  createList = (items) => {
    return items.map((i, id) => {
      return (<li key={id}><a href={i.href}>{i.title}</a></li>)
    })
  }

  render() {
    return (
      <div className={'project-tile-container'}>
        <div className={'project-tile-info'}>
          <div className={'project-tile-title'}>
            <h4>{this.props.title}</h4>
          </div>
          <div className={'project-tile-desc'}>
            <p>{this.props.description}</p>
          </div>
          <div className={'project-tile-links'}>
            <ul>
              {this.createList(this.props.links)}
            </ul>
          </div>
        </div>
        <img src={this.props.imgsrc} />
      </div>
    )
  }
}

ProjectTile.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imgsrc: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({
    href: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  }))
}
