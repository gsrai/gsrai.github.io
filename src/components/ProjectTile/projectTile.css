.card > img {
  max-height: 500px;
}

.project-links {
  margin-bottom: 1rem;
  margin-top: 2rem;
}