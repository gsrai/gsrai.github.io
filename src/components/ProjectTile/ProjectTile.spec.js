import { shallow } from 'enzyme'
import React from 'react'
import ProjectTile from './index'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const props = {
      title: 'Some Totally Awesome Project',
      description: 'I did some stuff.',
      imgsrc: 'assets/foo.jpeg',
      links: [{ href: 'some/url', title: 'Github' }]
    }
    const wrapper = shallow(<ProjectTile {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
