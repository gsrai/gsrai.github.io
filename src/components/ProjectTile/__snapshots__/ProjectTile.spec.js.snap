// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "marginBottom": "2rem",
      "width": "20rem",
    }
  }
>
  <img
    alt="Some Totally Awesome Project"
    src="assets/foo.jpeg"
    title="Some Totally Awesome Project"
  />
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      Some Totally Awesome Project
    </h4>
    <p
      className="card-text project-desc"
    >
      I did some stuff.
    </p>
    <div
      className="project-links"
    >
      <a
        className="paper-btn margin "
        href="some/url"
        key="0"
      >
        Github
      </a>
    </div>
  </div>
</div>
`;
